Update "Active Record Associations" guide [ci skip]

bogdanvlviv opened the pull request:
- Make all `ActiveRecord::Base.find` as link

- <del>Remove "It also adds the additional condition that the object must be in the collection."
  sentence for `collection.where` since it isn't correct statement.
  Remove the sentence for `collection.find` for consistency with others.</del>

 - Remove redundant sentences
    "It also adds the additional condition that the object must be in the collection."

kaspth commented:
Merely saying "Update" in the commit message doesn't cut it for me. Please elaborate what you've changed and why.

kaspth commented:
> It also adds the additional condition that the object must be in the collection.

What's wrong about this? `post.comments.where(…)` would still end up with `SELECT * FROM comments WHERE post_id = ?`, no?

bogdanvlviv committed:
Update "Active Record Associations" guide [ci skip]

  - Make all `ActiveRecord::Base.find` as link
  - Remove redundant sentences
    "It also adds the additional condition that the object must be in the collection."

bogdanvlviv commented:
@kaspth Thanks for the review. looks like I was misunderstanding the sentence `"It also adds the additional condition that the object must be in the collection"`. I just removed changes related to this.

eugeneius commented on guides/source/association_basics.md:
```diff
@@ -2091,7 +2092,9 @@ The `collection.size` method returns the number of objects in the collection.
-The `collection.find` method finds objects within the collection. It uses the same syntax and options as `ActiveRecord::Base.find`. It also adds the additional condition that the object must be in the collection.
+The `collection.find` method finds objects within the collection. It uses the same syntax and options as
+[`ActiveRecord::Base.find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find).
+It also adds the additional condition that the object must be in the collection.
```
> The `collection.find` method finds objects within the collection. [...] It also adds the additional condition that the object must be in the collection.

This seems totally redundant to me; I think you were right to remove it initially.

I think what it's _trying_ to say is that `collection.find` behaves identically to `ActiveRecord::Base.find`, except that the query will be scoped to the collection. I think that's clear enough from the first two sentences, though.

WDYT @kaspth?

kaspth replied:
Yes, the line is totally redundant and can be removed. @bogdanvlviv has expressed interest in upping his contributions, so I'm badgering him to explain his understanding and come up with better arguments for it.

In other words: the line shouldn't be removed because "it's wrong" (it ain't), but it should be removed because it's redundant. Bogdan, you know what to do 😊

eugeneius replied:
Apologies, I didn't recognise the Socratic method at work here. 😁

bogdanvlviv replied:
Done.

kaspth replied:
@eugeneius pretty hidden, so no worries! 😊

@bogdanvlviv cool 👍
